src/librustc_mir/transform/check_consts/ops.rs

```diff
@@ -268,18 +268,24 @@ impl NonConstOp for Panic {
 #[derive(Debug)]
 pub struct RawPtrComparison;
 impl NonConstOp for RawPtrComparison {
-    fn feature_gate() -> Option<Symbol> {
-        Some(sym::const_compare_raw_pointers)
-    }
-
     fn emit_error(&self, ccx: &ConstCx<'_, '_>, span: Span) {
-        feature_err(
-            &ccx.tcx.sess.parse_sess,
-            sym::const_compare_raw_pointers,
+        let mut err = ccx.tcx.sess.struct_span_err(
             span,
-            &format!("comparing raw pointers inside {}", ccx.const_kind()),
-        )
-        .emit();
+            "pointers cannot be compared in a meaningful way during const eval.",
+        );
+        err.note(
+            "It is conceptually impossible for const eval to know in all cases whether two \
+             pointers are equal. While sometimes it is clear (the address of a non-zst static item \
```

Comment:
```suggestion
             pointers are equal at runtime. While sometimes it is clear (the address of a non-zst static item \
```
Also shouldn't this be broken at 80 or 100 characters or so?

Reply:
Oh and do we usually capitalize ZST or not?

Reply:
Honestly, this is probably too much text. I think it makes more sense to just say "we cannot reliably determine whether two pointers are equal at compiletime, see URL for further information".

Reply:
With rustfmt that isn't enforced anymore, I fixed that

Reply:
I meant the output, not the source.

Reply:
Also even with your latest changes this is still a wall of text -- it is *by far* the largest diagnostic message I have ever seen. Wouldn't it make sense to just move all that information behind some URL? In particular the non-zst static example seems like way too much detail.

Reply:
I can move it behind `--teach`

Reply:
Sorry I only saw half of the comments here. Github didn't show me anything but the initial comment before my last comment when I wrote it

Reply:
> I can move it behind --teach

I never used that flag, so -- maybe? But the more detailed explanations are AFAIK usually put in markdown file or so?

Reply:
for now all the info is behind the linked issue, so let's just go with that

Reply:
LGTM. Might be worth updating the OP of that issue with some of the text we worked on here.

Reply:
done